diff --git a/drivers/of/Kconfig b/drivers/of/Kconfig
index bc07ad3..e9da9cf 100644
--- a/drivers/of/Kconfig
+++ b/drivers/of/Kconfig
@@ -113,6 +113,13 @@ config OF_OVERLAY
 	  While this option is selected automatically when needed, you can
 	  enable it manually to improve device tree unit test coverage.
 
+config OF_CONFIGFS
+        bool "Device Tree Overlay ConfigFS interface"
+        select CONFIGFS_FS
+        depends on OF_OVERLAY
+        help
+          Enable a simple user-space driven DT overlay interface.
+
 config OF_NUMA
 	bool
 
