--- /dev/null
+++ b/drivers/net/wireless/realtek/rtl8723cs/hal/phydm/rtl8703b/version_rtl8703b.h
@@ -0,0 +1,10 @@
+/*RTL8703B PHY Parameters*/
+/* 
+[Caution] 
+  Since 01/Aug/2015, the commit rules will be simplified. 
+  You do not need to fill up the version.h anymore, 
+  only the maintenance supervisor fills it before formal release.
+*/
+#define	RELEASE_DATE_8703B		20150915
+#define	COMMIT_BY_8703B			"BB_Luke"
+#define	RELEASE_VERSION_8703B	17
diff --git a/drivers/net/wireless/realtek/rtl8723cs/hal/phydm/txbf/halcomtxbf.c b/drivers/net/wireless/realtek/rtl8723cs/hal/phydm/txbf/halcomtxbf.c
new file mode 100644
index 00000000..146b6e01
