diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 9570f05d2a663..3f980b4cd32d9 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -927,4 +927,5 @@ dtb-$(CONFIG_MACH_SUN8I) += \
 	sun8i-h3-nanopi-r1.dtb \
 	sun8i-h3-nanopi-duo2.dtb \
+	sun8i-h3-zeropi.dtb \
 	sun8i-h3-orangepi-2.dtb \
 	sun8i-h3-orangepi-lite.dtb \
diff --git a/arch/arm/boot/dts/sun8i-h3-zeropi.dts b/arch/arm/boot/dts/sun8i-h3-zeropi.dts
new file mode 100644
index 0000000000000..4edee84316249
